lib/active_model/serializer/field.rb

```diff
@@ -38,12 +38,14 @@ def included?(serializer)
       private
 
       def condition_type
-        if options.key?(:if)
-          :if
-        elsif options.key?(:unless)
-          :unless
-        else
-          :none
+        @condition_type ||= begin
```

Comment:
I'd be interested in benchmarking this against as case statement checking `options.keys`.